### ISSUE TITLE
No gradients for bbox detector layers

### ISSUE THREAD

**Issue by ROYALBEFF** [facebookresearch/detr#420]
Hi, I'm using the pre-trained DETR model with ResNet50 backbone and want to compute the gradients of the loss function w.r.t the model parameters. So far everything works as expected, but however the gradients w.r.t. the parameters in the bounding box detector layers are not computed, even though I run the gradient computation in training mode and the parameters have `requires_grad = True`. I wonder if this is really the desired behaviour and if so why. If not I wonder how I can also get the gradients for those parameters as well.

Here is a small code snippet giving you an idea of what happens:
```{python}
# load
feature_extractor = DetrFeatureExtractor.from_pretrained('facebook/detr-resnet-50')
model = DetrForObjectDetection.from_pretrained('facebook/detr-resnet-50')

# load data
coco = CocoDetection(args.input, annFile=args.annotation, transform=transforms.ToTensor())

# extract features
images, targets = zip(*coco)
features = feature_extractor(images, return_tensor='pt')

for img, f, target in tqdm(zip(images, features, targets)):
    pixel_values = torch.from_numpy(f['pixel_values'][0][np.newaxis])
    pixel_mask = torch.from_numpy(f['pixel_mask'][0][np.newaxis])
    output = model(pixel_values, pixel_mask)
    logits = output.logits[0]
    bboxes = center_to_min(scale_bboxes(output.pred_boxes[0], img.shape))
    prob = torch.nn.functional.softmax(logits, dim=-1).data

    matching = hungarian(target, prob, bboxes)
    for t, p in matching:
        ...
        # use logits since cross entropy loss already computes softmax
        loss = CrossEntropyLoss(logits[p].reshape([1, -1]), torch.tensor([label], dtype=torch.long))

        model.train()
        model.zero_grad()
        # do not free computation graph
        loss.backward(retain_graph=True)
        
        # visualize graph
        make_dot(loss, params=dict(model.named_parameters())).render('graph', format='pdf')

        for name, param in model.named_parameters():
            if param.requires_grad and param.grad is None:
                print(name)
```

This outputs the following parameters for each iteration of the loop:
```
bbox_predictor.layers.0.weight
bbox_predictor.layers.0.bias
bbox_predictor.layers.1.weight
bbox_predictor.layers.1.bias
bbox_predictor.layers.2.weight
bbox_predictor.layers.2.bias
```

When visualising the graph, it can be seen that it doesn't include the above-listed parameters either.

[graph.pdf](https://github.com/facebookresearch/detr/files/6911175/graph.pdf)

Due to some other dependencies I rely on python 3.6 and pytorch 1.3.1
